### PR TITLE
A working WorkspaceSymbolsHandler

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
@rjmholt this comes after https://github.com/rjmholt/PowerShellEditorServices/pull/15

It’s a whole lot of copying from other files. I’m happy to walk through it with you

### CODE REVIEW

File: src/PowerShellEditorServices.Engine/Hosting/EditorServicesHost.cs [hunk 1]
```diff
@@ -219,12 +219,15 @@ public void StartLogging(string logFilePath, PsesLogLevel logLevel)
         {
             while (System.Diagnostics.Debugger.IsAttached)
             {
-                Console.WriteLine($"{System.Diagnostics.Process.GetCurrentProcess().Id}");
-                System.Threading.Thread.Sleep(2000);
+                Console.WriteLine($"{Process.GetCurrentProcess().Id}");
```
Comment (rjmholt) [inline src/PowerShellEditorServices.Engine/Hosting/EditorServicesHost.cs:222]: I've realised the better pattern is:

```csharp
Console.WriteLine($"PID: {Process.GetCurrentProcess().Id}");
while (!Debugger.IsAttached)
{
    Thread.Sleep(500);
}
```

File: src/PowerShellEditorServices.Engine/LanguageServer/OmnisharpLanguageServer.cs [hunk 2]
```diff
@@ -86,7 +86,7 @@ public async Task WaitForShutdown()
             out NamedPipeServerStream outPipe)
         {
             // .NET Core implementation is simplest so try that first
-            if (DotNetFacade.IsNetCore)
+            if (VersionUtils.IsNetCore)
```
Comment (rjmholt) [inline src/PowerShellEditorServices.Engine/LanguageServer/OmnisharpLanguageServer.cs:89]: I think `VersionUtils` is going to have narrow utility compared to something like `DotNetFacade` and `PowerShellFacade` where each wraps a common API with a different, complex or cached implementation that we want to use across the repo.

**Review by rjmholt — APPROVED** [review]

File: src/PowerShellEditorServices.Engine/LanguageServer/OmnisharpLanguageServer.cs (hunk 2 above)
Comment (TylerLeonhardt) [reply src/PowerShellEditorServices.Engine/LanguageServer/OmnisharpLanguageServer.cs:89]: Lets cross that bridge when we get there

File: src/PowerShellEditorServices.Engine/Hosting/EditorServicesHost.cs (hunk 1 above)
Comment (TylerLeonhardt) [reply src/PowerShellEditorServices.Engine/Hosting/EditorServicesHost.cs:222]: Yeah I typically just add that `!` in front of `Debugger` when I need to debug

Comment (TylerLeonhardt) [reply src/PowerShellEditorServices.Engine/Hosting/EditorServicesHost.cs:222]: of course, this code will be gone once it goes in

**Pull request closed without merging**